src/reporting.ts

```diff
@@ -54,26 +55,38 @@ export function createIssueCommentMessage(issue: IssueOccurrence): string {
   const message = createReviewCommentMessage(issue)
   const relativePath = relativizePath(issue.mainEventFilePathname)
 
+
   return `${message}
 ## Issue location
 This issue was discovered outside the diff for this Pull Request. You can find it at:
 [${relativePath}:${issue.mainEventLineNumber}](${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}/blob/${process.env.GITHUB_SHA}/${relativePath}#L${issue.mainEventLineNumber})
 `
 }
 
+// naive Z-Table, can be optimized to terminate early
+function zTable(input: string) {
+    let table = new Array<number>(input.length).fill(0)
+    for (let i = 1; i < input.length; i++) {
+        for (let j = 0; j < input.length - i; j ++) {
+            if (input.charAt(j) != input.charAt(i + j)) continue
+            table[i]++
+        }
+    }
+    return table
+}
```

Comment:
this can be optimized by terminating early but not really necessary since path are tiny